gateway-service/src/test/java/org/zowe/apiml/gateway/filters/pre/LocationFilterTest.java

```diff
@@ -85,7 +88,7 @@ void urlDefinedInMetadataIsModified() {
     void testContextKeyURLFiltering(String contextKey, String contextUrl, String requestUrl) {
         final RequestContext ctx = RequestContext.getCurrentContext();
         ctx.set(contextKey, contextUrl);
-        this.filter.run();
+        if (this.filter.shouldFilter()) this.filter.run();
```

Comment:
I'm really hesitant about an if statement like this in a test, it breaks tests isolation. Could this test be split into 2 tests, one testing `shouldFilter` and one testing `run`, with appropriate inputs for each?